.github/workflows/publish.yaml

```diff
@@ -8,19 +8,28 @@ jobs:
   build:
     name: Build distribution package
     runs-on: ubuntu-latest
-
     steps:
       - uses: actions/checkout@v4
         with:
           persist-credentials: false
+      - name: Fetch tags
+        run: git fetch --prune --unshallow --tags
       - name: Set up Python
         uses: actions/setup-python@v5
         with:
           python-version: "3.x"
       - name: Install pypa/build
-        run: python -m pip install --user build
+        run: python -m pip install --user build twine
+      - name: Add '.devN' to version for non-tag builds
+        if: ${{ ! startsWith(github.ref, 'refs/tags/') }}
+        run:
+          sed -i
+          "/^APP_VERSION = /s/'$/.dev$(git describe --tags | cut -d- -f2)'/"
+          yamllint/__init__.py
       - name: Build a binary wheel and a source tarball
         run: python -m build
+      - name: Twine check the distribution packages
+        run: python -m twine check --strict dist/yamllint-*
```

Comment:
@webknjaz `python -I` (that you suggested at https://github.com/adrienverge/yamllint/issues/721#issuecomment-2741928694) didn't work  because in isolated mode, `python` doesn't find user-installed libraries:

   /opt/hostedtoolcache/Python/3.13.2/x64/bin/python: No module named twine

Reply:
Oh, I didn't realize. In general, it's discouraged to use `--user` because it's a global location with some things (like installed binaries) shared across several Python interpreters which may override each other.
I should really drop that from the guide… Nowadays, I default to having `-I` in all (or most of) my `python` invocations for safety reasons.

Technically, GH-provided VMs have `pipx` installed. So you could get away with just having `pipx run build` followed by a `pipx run twine check --strict dists/*`. If you want to pin versions, you can add `--spec` there.